src/Obj.hs

```diff
@@ -1092,11 +1092,11 @@ wrapInRefTyIfMatchRef MatchValue t = t
 isResolvableStaticObj :: Obj -> Bool
 isResolvableStaticObj Def = True
 isResolvableStaticObj (Defn _) = True
-isResolvableStaticObj (External _) = True
-isResolvableStaticObj (Deftemplate _) = True
-isResolvableStaticObj (Instantiate _) = True
-isResolvableStaticObj (Fn _ _) = True
 isResolvableStaticObj (Interface _ _) = True
+isResolvableStaticObj (Instantiate _) = True
+isResolvableStaticObj (Deftemplate _) = True
+isResolvableStaticObj (External _) = True
+isResolvableStaticObj (Match _) = True
```

Comment:
There's some context in commit msg [5cc5752](https://github.com/carp-lang/Carp/pull/1385/commits/5cc575237f4b204c8576beeddf55e4362a662342) -- basically there used to be multiple approaches to checking static calls, two different functions in `eval` and one direct use of `isResolvableObj`. I merged the two different definitions of `checkStatic` by making them rely on `isResolvableObj`--however, this caused failures in macros when `Fn`s were encountered since they were checked by isResolvable. Removing that check doesn't seem to break anything, so I dropped it.

Reply:
Also, the addition of Match is similar -- merged to bring parity with the two old separate definitions.